feat:lesson05

nakagawamai opened the pull request:
# [Issue5](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#5)
4のこちらをつかって、こんどはPromiseオブジェクトを使って解決された値として受け取り、
```
[{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
````
同じように
````
<ul>
 <li><a href="/bookmark.html"><img src="1.png" alt="画像1">ブックマーク</a></li>
 <li><a href="/message.html"><img src="2.png" alt="画像2">メッセージ</a></li>
</ul>
````
こちらの出力になるようにしてください
## [CodeSandBox](https://codesandbox.io/s/icy-night-nc9z31?file=/script.js) :[Latest Link](https://github.com/nakagawamai/js-lesson/pull/6/commits/18b98169e4788113840bc31b988242829eb5944c)

## Uneasy part
- Does the meaning change if the loop processing is performed first?

nakagawamai committed:
feat:lesson05

use promise object and receive as a resolved value and create a DOM.

nakagawamai commented on lesson05/script.js:
```diff
@@ -0,0 +1,25 @@
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+   ]
+
+const promise = new Promise(resolve => {
+    resolve(lists)
+});
+
+promise.then(() => {
+    const fragment = document.createDocumentFragment();
+    for(const list of lists){
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = '/' +list.to;
+        img.src = list.img;
+        img.alt = list.alt;
+
+        fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
+    }
+    ul.appendChild(fragment);
+});
```
Does the meaning change if I first run the loop process to create the DOM?
````
const promise = new Promise(resolve => {
    resolve(lists)
});

const fragment = document.createDocumentFragment();
for(const list of lists){
    const li = document.createElement('li');
    const a = document.createElement('a');
    const img = document.createElement('img');
    a.textContent = list.text;
    a.href = '/' +list.to;
    img.src = list.img;
    img.alt = list.alt;

    fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
}

promise.then(() => {
    ul.appendChild(fragment);
});
````

seeeen044 commented on lesson05/script.js:
```diff
@@ -0,0 +1,25 @@
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+        a.textContent = list.text;
+        a.href = '/' +list.to;
```
```suggestion
        a.href = `/${list.to}`;
```
nits
You can also write it this way.
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Template_literals

nakagawamai replied:
Thank you for your comment!
I study template literals. 
I will write using template literals.

csakyo commented on lesson05/script.js:
```diff
@@ -0,0 +1,25 @@
+   ]
+
+const promise = new Promise(resolve => {
+    resolve(lists)
```
MUST
You need to put the ';' after ().
``resolve(lists);``

nakagawamai replied:
Thank you for teaching me.
I forgot to add the ";" at the end.

csakyo commented on lesson05/script.js:
```diff
@@ -0,0 +1,25 @@
+    resolve(lists)
+});
+
+promise.then(() => {
```
IMO
I think you should set the argument here. 
``promise.then((lists) => {...``
This is because we need to receive the resolved values here.
Now, when I turn off the value of resolve``resolve()``, it still works, so maybe you have referenced the object globally. The resolved value is not being used.

nakagawamai replied:
I made a mistake!
The value received by resolve should be stored in a variable.

I need to write:
`promise.then. ((lists) => {...`

seeeen044 commented on lesson05/script.js:
```diff
@@ -0,0 +1,25 @@
+
+promise.then(() => {
+    const fragment = document.createDocumentFragment();
+    for(const list of lists){
```
Must

I overlooked an important place. I'm sorry.

If it's a `const list of lists `, I'm going to refer to `const lists`instead of receiving the resolved value. What should I do?

nakagawamai replied:
Thank you for your comment!

I was very misunderstood.
I had to store the value received by the resolve in a variable.

nakagawamai committed:
fix:Added variable to receive value.

I forgot to write the variable to store the value received by resolve, so I corrected it.

nakagawamai commented on lesson05/script.js:
```diff
@@ -5,17 +5,17 @@ const lists = [
 
 const promise = new Promise(resolve => {
-    resolve(lists)
+    resolve(lists);
```
I forgot to write the ";", so I added it.

nakagawamai commented on lesson05/script.js:
```diff
@@ -5,17 +5,17 @@ const lists = [
-promise.then(() => {
+promise.then((lists) => {
     const fragment = document.createDocumentFragment();
     for(const list of lists){
```
Added variable to receive value.
Because we need to prepare a variable to store the value received by resolve.

nakagawamai commented on lesson05/script.js:
```diff
@@ -5,17 +5,17 @@ const lists = [
         const img = document.createElement('img');
         a.textContent = list.text;
-        a.href = '/' +list.to;
+        a.href = `/${list.to}`;
```
I use template literals.

nakagawamai committed:
fix:Fixed loop reference.

Fixed because the reference part of the loop processing was wrong.

nakagawamai commented on lesson05/script.js:
```diff
@@ -8,9 +8,9 @@ const promise = new Promise(resolve => {
 });
 
-promise.then((lists) => {
+promise.then((value) => {
     const fragment = document.createDocumentFragment();
-    for(const list of lists){
+    for(const list of value){
         const li = document.createElement('li');
         const a = document.createElement('a');
         const img = document.createElement('img');
```
Fixed because the reference part of the loop processing was wrong.

nakagawamai replied:
I checked the value of "value".

Result of `console.log (value);`
````
0: {to: 'bookmark.html', img: '1.png', alt: '画像1', text: 'ブックマーク'}
1: {to: 'message.html', img: '2.png', alt: '画像2', text: 'メッセージ'}
length: 2
